lib/barbeque/execution_log.rb

```diff
@@ -7,35 +7,54 @@ class ExecutionLog
     DEFAULT_S3_BUCKET_NAME = 'barbeque'
 
     class << self
-      delegate :save, :load, to: :new
+      delegate :save_message, :save_stdout_and_stderr, :load, to: :new
 
       def s3_client
         @s3_client ||= Aws::S3::Client.new
       end
     end
 
+    # @param [Barbeque::JobExecution] execution
+    # @param [Barbeque::Message::JobExecution] message
+    def save_message(execution, message)
+      put(execution, 'message.json', message.body.to_json)
+    end
+
     # @param [Barbeque::JobExecution,Barbeque::JobRetry] execution
-    # @param [Hash] log
-    def save(execution:, log:)
-      ExecutionLog.s3_client.put_object(
-        bucket: s3_bucket_name,
-        key:    s3_key_for(execution: execution),
-        body:   log.to_json,
-      )
+    # @param [String] stdout
+    # @param [String] stderr
+    def save_stdout_and_stderr(execution, stdout, stderr)
+      put(execution, 'stdout.txt', stdout)
+      put(execution, 'stderr.txt', stderr)
     end
 
     # @param [Barbeque::JobExecution,Barbeque::JobRetry] execution
     # @return [Hash] log
     def load(execution:)
       return {} if execution.pending?
 
-      s3_object = ExecutionLog.s3_client.get_object(
-        bucket: s3_bucket_name,
-        key:    s3_key_for(execution: execution),
-      )
-      JSON.load(s3_object.body.read)
-    rescue Aws::S3::Errors::NoSuchKey
-      nil
+      message_str = get(execution, 'message.json')
+      message = message_str ? JSON.parse(message_str) : nil
```

Comment:
Parsing JSON was unnecessary because the previous implementation saved log with `{ message: @message.body.to_json, stdout: stdout, stderr: stderr }.to_json`
https://github.com/cookpad/barbeque/commit/e15c57c4cdfb855ee6c4524fa5d04de7412a51f0